pkg/controller/util.go

```diff
@@ -215,6 +216,77 @@ func CreateScratchPersistentVolumeClaim(client client.Client, pvc *v1.Persistent
 	return scratchPvc, nil
 }
 
+// GetStorageClassByName looks up the storage class based on the name. If no storage class is found returns nil
+func GetStorageClassByName(client client.Client, name *string) (*storagev1.StorageClass, error) {
+	// look up storage class by name
+	if name == nil {
+		storageClasses := &storagev1.StorageClassList{}
+		if err := client.List(context.TODO(), storageClasses); err != nil {
+			klog.V(3).Info("Unable to retrieve available storage classes")
+			return nil, errors.New("unable to retrieve storage classes")
+		}
+		for _, storageClass := range storageClasses.Items {
+			if storageClass.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
+				return &storageClass, nil
+			}
+		}
+	} else {
+		storageClass := &storagev1.StorageClass{}
+		if err := client.Get(context.TODO(), types.NamespacedName{Name: *name}, storageClass); err != nil {
+			klog.V(3).Info("Unable to retrieve storage class", "storage class name", *name)
+			return nil, errors.New("unable to retrieve storage class")
+		}
+		return storageClass, nil
+	}
+	// No storage class found, just return nil for storage class and let caller deal with it.
+	return nil, nil
+}
+
+// GetFilesystemOverhead determines the filesystem overhead defined in CDIConfig for this PVC's volumeMode and storageClass.
+func GetFilesystemOverhead(client client.Client, pvc *v1.PersistentVolumeClaim) (cdiv1.Percent, error) {
+	klog.V(1).Info("GetFilesystemOverhead with PVC", pvc)
+	if getVolumeMode(pvc) != v1.PersistentVolumeFilesystem {
+		return "0", nil
+	}
+
+	cdiConfig := &cdiv1.CDIConfig{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: common.ConfigName}, cdiConfig); err != nil {
+		if k8serrors.IsNotFound(err) {
+			klog.V(1).Info("CDIConfig does not exist, pod will not start until it does")
+			return "0", nil
+		}
+
+		return "0", err
+	}
+
+	targetStorageClass, err := GetStorageClassByName(client, pvc.Spec.StorageClassName)
+	if err != nil {
+		klog.V(1).Info("Storage class", pvc.Spec.StorageClassName, "not found, trying default storage class")
+		targetStorageClass, err = GetStorageClassByName(client, nil)
```

Comment:
You populate err here, but the rest of the function never uses it. If you don't care about the err use _, or if you do care, check the value and do something with it. If you don't care explain in a comment why not.